source/grid/manifold_lib.cc

```diff
@@ -281,71 +346,151 @@ get_new_point (const ArrayView<const Point<spacedim>> &vertices,
 {
   const unsigned int n_points = vertices.size();
 
-  double rho = 0.0;
-  Tensor<1,spacedim> candidate;
-  for (unsigned int i = 0; i<n_points; i++)
-    {
-      const Tensor<1,spacedim> direction (vertices[i]-center);
-      rho += direction.norm()*weights[i];
-      candidate += direction*weights[i];
-    }
+  if (n_points == 2)
+    return get_intermediate_point(vertices[0], vertices[1],
+                                  std::max(0., std::min(weights[1],1.)));
```

Comment:
There is one catch with this change: Since we do use `SphericalManifold` also with some negative weights (e.g. when inside the Newton iteration of `TransfiniteInterpolationManifold`), you need to remove the check `w>=0 && w<=1` in line 241 at the top of `get_intermediate_point`. I have checked and it is uncritical because we nowhere assume that we look for a point in between. The result might be weird, but that's intended for uses in a context with negative weights (and others larger than 1).